pkg/client/core/service_application_client.go

```diff
@@ -127,6 +113,165 @@ func (s *applicationService) CreateArgoCDApplicationManifest(opts client.CreateA
 	return nil
 }
 
+// DeleteArgoCDApplicationManifest removes necessary files related to the ArgoCD integration
+// This function adds a finalizer to the ArgoCD application manifest before deletion, ensuring a cascading delete.
+// https://argo-cd.readthedocs.io/en/stable/user-guide/app_deletion/#about-the-deletion-finalizer
+func (s *applicationService) DeleteArgoCDApplicationManifest(opts client.DeleteArgoCDApplicationManifestOpts) error {
+	relativeArgoCDApplicationManifestPath := getRelativeArgoCDManifestPath(opts.Cluster, opts.Application)
+	absoluteArgoCDApplicationManifestPath := path.Join(s.absoluteRepositoryDir, relativeArgoCDApplicationManifestPath)
+
+	err := s.deleteFileInRemote(opts.Cluster, opts.Application, relativeArgoCDApplicationManifestPath)
+	if err != nil {
+		return fmt.Errorf("deleting application manifest in repository: %w", err)
+	}
+
+	err = s.fs.Remove(absoluteArgoCDApplicationManifestPath)
+	if err != nil {
+		if stderrors.Is(err, os.ErrNotExist) {
+			return nil
```

Comment:
If I read this correctly: if we try to remove a manifest, and it doesn't exist, we just return nil? Is this in case we run this operation several times, and we have already removed it?

I see we have several of this `nil` return cases in this patch, so I guess it is all about having done it before. But what if it is the wrong input to this function? Or something changes, wouldn't this just silence any error?

Reply:
Hmm good point. It's supposed to ensure that all parts get handled if something crashes mid action. I see theres faults in the logic writes code